### DISCUSSION TITLE
Supported cameras for manual focus

### DISCUSSION THREAD

**Discussion by Ultrawipf** [frank26080115/alpha-fairy#22] [Q&A]
Currently testing this with a A6400 with a SELP18105G lens and while the shutter features work, anything involving focus adjustments reports "unsupported camera".
In the official Sony list of available API commands manual focus is not listed for any camera and imaging edge also does not have this feature but i assume at least for the A7 line it should support focus. 
Seems very strange to me supporting zoom but not manual focus via the API...

Can someone confirm on which cameras the focus functions work?

**Comment by frank26080115** [comment]
You are correct about examining the old SDK document to see what commands are supported. Sorry I can't do anything about the A6400 not having that function.

Any camera that supports the PTP protocol SDK should support manual focusing, check the supported device list here: https://support.d-imaging.sony.co.jp/app/sdk/en/index.html

As for confirmation from users, look at what we are discussing in the github issues (closed ones too). And I'm always glad if somebody simply makes a post stating a certain camera works.

> Seems very strange to me supporting zoom but not manual focus via the API...

The kit lens for all the cameras since the NEX camera days all have power-zoom, plus the RX100 series. Plus, nobody uses MF on smartphones so nobody knows what MF actually is. I think these are the reasons. Only speculating.

**Reply by Ultrawipf** [reply]
It does support PTP but the new SDK does not list any 6000 series. 
Tested the demo application anyways and as expected it does not detect it and the detection function is hidden in a dll.
And the old API lists the 6300 and 6500 but not the newer 6400 but it is mostly the same.
Thats really annoying they crippled the functionality of the APS-C models that much but thanks for confirming that its probably not implemented by Sony...

> The kit lens for all the cameras since the NEX camera days all have power-zoom, plus the RX100 series.

But they also all have at least some manual focus options so its not unreasonable to expect that function via the API as well. I wouldn't call MF an obscure option. Even on phones its sometimes essential to use even if not everybody uses it. But i think enough people do that it would add value to the products.
